prompt2model/model_trainer/generate.py

```diff
@@ -32,6 +34,7 @@ def __init__(
         executor_batch_size: int = 10,
         tokenizer_max_length: int = 512,
         sequence_max_length: int = 1024,
+        device: Optional[str] = None,
```

Comment:
We prefer to use union types, which would set this as:
```suggestion
        device: str | None = None,
```

To support this, you will also have to add
`from __future__ import annotations  # noqa FI58`

In the first line of this file.